core/common/src/main/java/alluxio/PropertyDocGeneration.java

```diff
@@ -0,0 +1,138 @@
+    fileNames.put("master", filePath + "master-configuration.csv");
+    fileNames.put("worker", filePath + "worker-configuration.csv");
+    fileNames.put("security", filePath + "security-configuration.csv");
+    fileNames.put("keyvalue", filePath + "key-value-configuration.csv");
+    fileNames.put("common", filePath + "common-configuration.csv");
+
+    //HashMap for FileWriter per each category
+    Map<String, FileWriter> fileWriterMap = new HashMap<>();
+    try {
+      for (Map.Entry<String, String> entry : fileNames.entrySet()) {
+        fileWriter = new FileWriter(entry.getValue());
+        //Write the CSV file header
+        fileWriter.append(fileHeader);
+        //Add a new line separator after the header
+        fileWriter.append("\n");
+        //put fileWriter
+        fileWriterMap.put(entry.getKey(), fileWriter);
+      }
+
+      for (Map.Entry<PropertyKey, Object> entry : defaultVals.entrySet()) {
+        String pKey = entry.getKey().toString();
+        String value;
+        if (entry.getValue() == null) {
+          value = "";
+        } else {
+          value = entry.getValue().toString();
+        }
+
+        //Persist property key and default value to CSV
+        String keyValueStr = pKey + "," + value + "\n";
+        if (pKey.contains(".user.")) {
+          fileWriter = fileWriterMap.get("user");
+          fileWriter.append(keyValueStr);
+        } else if (pKey.contains(".master.")) {
+          fileWriter = fileWriterMap.get("master");
+          fileWriter.append(keyValueStr);
+        } else if (pKey.contains(".worker.")) {
+          fileWriter = fileWriterMap.get("worker");
+          fileWriter.append(keyValueStr);
+        } else if (pKey.contains(".security.")) {
+          fileWriter = fileWriterMap.get("security");
+          fileWriter.append(keyValueStr);
+        } else if (pKey.contains(".keyvalue.")) {
+          fileWriter = fileWriterMap.get("keyvalue");
+          fileWriter.append(keyValueStr);
+        } else {
+          fileWriter = fileWriterMap.get("common");
+          fileWriter.append(keyValueStr);
+        }
+      }
+
+      LOG.info("Property Key CSV files were created successfully.");
+    } catch (Exception e) {
+      LOG.error("Error in Property Key CSV FileWriter", e);
+    } finally {
+      try {
+        for (Map.Entry<String, FileWriter> entry : fileWriterMap.entrySet()) {
+          entry.getValue().flush();
+          entry.getValue().close();
+        }
+      } catch (IOException e) {
+        LOG.error("Error while flushing/closing Property Key CSV FileWriter", e);
+      }
+    }
+  }
+
+  /**
+   * Function for main entry.
+   *
+   * @param args arguments for command line
+   */
+  public static void main(String[] args) throws IllegalAccessException, NoSuchFieldException {
+    PropertyKey propertyKey = new PropertyKey("");
+    Class pk = propertyKey.getClass();
+    try {
+      Field defaultValues = pk.getDeclaredField("DEFAULT_VALUES");
```

Comment:
instead of using reflection, can we use `PropertyKey.defaultKeys()` to get the set of all pre-defined keys,
then pass this set of keys to `writeCSVFile`? Inside `writeCSVFile`, given a key, we can call `getDefaultValue` to get the value of this key